src/dmd/mars.d

```diff
@@ -1777,6 +1777,23 @@ bool parseCommandLine(const ref Strings arguments, const size_t argc, ref Param
             params.showColumns = true;
         else if (arg == "-vgc") // https://dlang.org/dmd.html#switch-vgc
             params.vgc = true;
+	else if (startsWith(p + 1, "recursion-limit"))
+	{
+	    enum STRING_SIZE = "-recursion-limit".length;
+	    if (p[STRING_SIZE] != '=')
+	    {
+		errorInvalidSwitch(p, "Expected argument following `-recursion-limit , e.g. `-recursion-limit=1000`");
+		return true;
+	    }
+	    if (!params.recursionLimit.parseDigits(p.toDString()[STRING_SIZE + 1 .. $]))
+	    {
+                errorInvalidSwitch(p, "Only numbers are allowed for -recursion-limit`");
+	    }
+	    if( params.recursionLimit == 0)
```

Comment:
Styling needs adjusting here.

Reply:
You are right, I am sorry (vim tabs misleaded me).
Corrected.